src/libertem/common/executor.py

```diff
@@ -107,6 +115,38 @@ def enter(self):
             else:
                 yield self
 
+    @contextmanager
+    def use_gpu(self, enable: bool = True):
+        """
+        Use the GPU specified in the constructor in CuPy
+
+        .. versionadded :: 0.16.0
+
+        Parameters
+        ----------
+
+        enable : bool, default True
+            Flag that allows using the GPU conditionally without changing the
+            code structure. If set to :code:`False`, this function doesn't
+            perform any action.
+        """
+        if enable:
+            with use_gpu(self._gpu_id):
+                yield
+        else:
+            yield
```

Comment:
What do you think about not having a separate `use_gpu` contextmanager, but instead having that as part of `enter`? Before, this class had one specific use: you specify the environment with the parameters of the constructor, and you `enter` it. Now, there are basically two environments you can specify with different parameters of the constructor, and two context managers. So I would propose to either 1) integrate into `enter` or 2) have a separate class altogether.